src/backup/backup.c

```diff
@@ -103,8 +103,8 @@ static backup_error_t _check_backup(uint8_t* output, size_t output_length, const
  */
 backup_error_t backup_create(uint32_t backup_create_timestamp, uint32_t seed_birthdate_timestamp)
 {
-    Backup __attribute__((__cleanup__(backup_cleanup_backup))) backup;
-    BackupData __attribute__((__cleanup__(backup_cleanup_backup_data))) backup_data;
+    Backup __attribute__((__cleanup__(backup_cleanup_backup))) backup = {0};
+    BackupData __attribute__((__cleanup__(backup_cleanup_backup_data))) backup_data = {0};
```

Comment:
@Beerosagos any idea why thse are zero initialized even without the ` = {0}`? Afaik it should be undefined and contain garbage, but it seems it's always zero anyway.

If it contained garbage, then backup_fill() would only fill the first e.g. 16 bytes of the 32 bytes-sized `seed` field if the created seed is 16 bytes (e.g. if you restore from a 12 word mnemonic phrase and then create an backup onto SD), but it seems that didn't happen.